src/neptune_query/internal/composition/fetch_table.py

```diff
@@ -127,31 +131,111 @@ def go_fetch_sys_attrs() -> Generator[list[identifiers.SysId], None, None]:
                 ),
             ),
         )
-        results: Generator[util.Page[att_vals.AttributeValue], None, None] = concurrency.gather_results(output)
 
-        for result in results:
-            attribute_values_page = result
+        attribute_value_pages: Generator[util.Page[att_vals.AttributeValue], None, None] = concurrency.gather_results(
+            output
+        )
+        for attribute_values_page in attribute_value_pages:
             for attribute_value in attribute_values_page.items:
-                sys_id = attribute_value.run_identifier.sys_id
-                result_by_id[sys_id].append(attribute_value)
+                result_by_id[attribute_value.run_identifier.sys_id].append(attribute_value)
 
-    result_by_name = _map_keys_preserving_order(result_by_id, sys_id_label_mapping)
-    dataframe = output_format.convert_table_to_dataframe(
-        table_data=result_by_name,
-        project_identifier=project_identifier,
+    return create_runs_table(
+        table_rows=[
+            TableRow(values=values, label=sys_id_label_mapping[sys_id], project_identifier=project_identifier)
+            for sys_id, values in result_by_id.items()
+        ],
         type_suffix_in_column_names=type_suffix_in_column_names,
-        index_column_name="experiment" if container_type == search.ContainerType.EXPERIMENT else "run",
+        container_type=container_type,
     )
 
-    return dataframe
 
+def fetch_table_global(
+    *,
+    filter_: Optional[_Filter],
+    attributes: _BaseAttributeFilter,
+    sort_by: _Attribute,
+    sort_direction: Literal["asc", "desc"],
+    limit: Optional[int],
+    type_suffix_in_column_names: bool,
+    context: Optional[_context.Context] = None,
+    container_type: search.ContainerType,
+) -> pd.DataFrame:
+    validation.validate_limit(limit)
+    _sort_direction = validation.validate_sort_direction(sort_direction)
+
+    if sort_by.type is None:
+        raise NeptuneUserError(
+            "Cannot resolve sort_by attribute type. Please pass filters.Attribute with an explicit type."
+        )
+
+    valid_context = _context.validate_context(context or _context.get_context())
+    client = _client.get_client(context=valid_context)
+
+    # Order is preserved here
+    table_rows: dict[identifiers.RunIdentifier, TableRow] = {}
```

Comment:
change to defaultdict, please

Reply:
I removed setdefault, leaving a normal dict. And fixed a bug, by the way.